test/lib/server/services/qualityControlFlag/QcFlagService.test.js

```diff
@@ -230,4 +230,356 @@ module.exports = () => {
             expect(fetchedSortedProperties).to.have.all.ordered.members([...fetchedSortedProperties].sort().reverse());
         });
     });
+
+    describe('Creating Quality Control Flag for data pass', () => {
+        /** Flags for runNumber: 106, LHC22b_apass1, CPV */
+        // Run trg time middle point: 1565314200, radius: 45000 seconds
+        it('should fail to create quality control flag due to incorrect external user id', async () => {
+            const qcFlagCreationParameters = {
+                from: new Date('2019-08-09 01:29:50').getTime(),
+                to: new Date('2019-08-09 05:40:00').getTime(),
+                comment: 'VERY INTERSETING REMARK',
+            };
+
+            const relations = {
+                user: {
+                    externalUserId: 9999999, // Failing property
+                },
+                flagTypeId: 2,
+                runNumber: 106,
+                dataPassId: 1,
+                dplDetectorId: 1,
+            };
+
+            await assert.rejects(
+                () => qcFlagService.createForDataPass(qcFlagCreationParameters, relations),
+                new BadParameterError('User with this external id (9999999) could not be found'),
+            );
+        });
+
+        it('should fail to create quality control flag due to incorrect qc flag time period', async () => {
```

Comment:
More precise test name please, the next one has the same. how is the time period incorrect?

Reply:
applied